From a03c21ae615b2f685f978b870526c37d3684b039 Mon Sep 17 00:00:00 2001
From: Bogdan Pricop <bogdan.pricop@emutex.com>
Date: Thu, 5 Feb 2015 17:29:15 +0000
Subject: [PATCH 239/389] tty: serial: sc16is7xx: fix Rx/Tx for the second
 port.

Add port number to SPI message sent to device. This fixes an
error communicating with dual-port variants such as the
SC16IS752 and SC16IS762

Signed-off-by: Bogdan Pricop <bogdan.pricop@emutex.com>
Signed-off-by: Dan O'Donovan <dan@emutex.com>
Signed-off-by: Kartikey Parmar <kartikeyx.r.parmar@intel.com>
---
 drivers/tty/serial/sc16is7xx.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/tty/serial/sc16is7xx.c b/drivers/tty/serial/sc16is7xx.c
index 8115fbf..4e73f55 100644
--- a/drivers/tty/serial/sc16is7xx.c
+++ b/drivers/tty/serial/sc16is7xx.c
@@ -496,9 +496,10 @@ static void sc16is7xx_handle_rx(struct uart_port *port, unsigned int rxlen,
 			s->buf[0] = sc16is7xx_port_read(port, SC16IS7XX_RHR_REG);
 			bytes_read = 1;
 		} else {
+			u8 reg = SC16IS7XX_RHR_REG << SC16IS7XX_REG_SHIFT |
+				port->line;
 			regcache_cache_bypass(s->regmap, true);
-			regmap_raw_read(s->regmap, SC16IS7XX_RHR_REG,
-					s->buf, rxlen);
+			regmap_raw_read(s->regmap, reg,	s->buf, rxlen);
 			regcache_cache_bypass(s->regmap, false);
 			bytes_read = rxlen;
 		}
@@ -567,6 +568,8 @@ static void sc16is7xx_handle_tx(struct uart_port *port)
 	/* Get length of data pending in circular buffer */
 	to_send = uart_circ_chars_pending(xmit);
 	if (likely(to_send)) {
+		u8 reg = SC16IS7XX_THR_REG << SC16IS7XX_REG_SHIFT | port->line;
+
 		/* Limit to size of TX FIFO */
 		txlen = sc16is7xx_port_read(port, SC16IS7XX_TXLVL_REG);
 		to_send = (to_send > txlen) ? txlen : to_send;
@@ -580,7 +583,7 @@ static void sc16is7xx_handle_tx(struct uart_port *port)
 			xmit->tail = (xmit->tail + 1) & (UART_XMIT_SIZE - 1);
 		}
 		regcache_cache_bypass(s->regmap, true);
-		regmap_raw_write(s->regmap, SC16IS7XX_THR_REG, s->buf, to_send);
+		regmap_raw_write(s->regmap, reg, s->buf, to_send);
 		regcache_cache_bypass(s->regmap, false);
 	}
 
-- 
1.9.3

